storage/storage.go

```diff
@@ -0,0 +1,56 @@
+package storage
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/go-openapi/runtime"
+	"github.com/sul-dlss-labs/taco/config"
+)
+
+// Storage the interface for file storage
+type Storage interface {
+	UploadFile(id string, file runtime.File) (*string, error)
+}
+
+// S3BucketStorage provides file storage using an S3 bucket
+type S3BucketStorage struct {
+	config   *config.Config
+	uploader *s3manager.Uploader
+}
+
+// NewS3Bucket creates a new S3 bucket storage
+func NewS3Bucket(config *config.Config) Storage {
```

Comment:
Do we need to create a new S3 Bucket here? I think we rather have the S3 bucket defined before and TACO would not be created new buckets, just passing files to the existing. Also, I don't think this method is use (afaict) so maybe just remove to make clear TACO shouldn't be creating new buckets.

Reply:
This doesn't create a new bucket in s3, it creates a new instance of S3Bucket which has the method to upload a file.

Reply:
gotcha, thanks for the clarification